lib/selinux.c

```diff
@@ -53,28 +53,40 @@ static bool selinux_enabled;
  */
 int set_selinux_file_context (const char *dst_name)
 {
-	/*@null@*/char *scontext = NULL;
-
 	if (!selinux_checked) {
 		selinux_enabled = is_selinux_enabled () > 0;
 		selinux_checked = true;
 	}
 
 	if (selinux_enabled) {
 		/* Get the default security context for this file */
-		if (matchpathcon (dst_name, 0, &scontext) < 0) {
-			if (security_getenforce () != 0) {
-				return 1;
+
+		/*@null@*/char *fcontext_raw = NULL;
+		struct selabel_handle *hnd;
+		int r;
+
+		hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
+		if (hnd == NULL) {
+			return security_getenforce () != 0;
+		}
+
+		r = selabel_lookup_raw(hnd, &fcontext_raw, dst_name, 0);
+		selabel_close(hnd);
+		if (r < 0) {
+			/* No context specified for the searched path */
+			if (errno == ENOENT) {
```

Comment:
Is selabel_close() implemented in a way that ensures that the errno from the failed selabel_lookup_raw() is not overwritten?

Reply:
I see - in the end it doesn't matter because you're removing that call.  Feel free to ignore my previous question about selabel_close().

Reply:
Good catch.
Should I add errno-saving code to the commit "selinux.c: use modern selabel interface instead of deprecated matchpathcon" and or squash "selinux.c: use modern selabel interface instead of deprecated matchpathcon" and "selinux: only open selabel database once", so that there is no intermediate state with this issue?

Reply:
Adding the errno-saving code would be overkill here :)  You can squash them, or ignore it.  Thanks.